src/boost_histogram/_internal/hist.py

```diff
@@ -376,24 +385,38 @@ def _reduce(self, *args):
         return self._new_hist(self._hist.reduce(*args))
 
     def __copy__(self):
-        other = self._new_hist(copy.copy(self._hist))
-        return other
+        return self._new_hist(copy.copy(self._hist))
 
     def __deepcopy__(self, memo):
-        other = self.__class__.__new__(self.__class__)
-        other._hist = copy.deepcopy(self._hist, memo)
-        other.metadata = copy.deepcopy(self.metadata, memo)
-        other.axes = other._generate_axes_()
-        return other
+        return self._new_hist(copy.deepcopy(self._hist), memo=memo)
 
     def __getstate__(self):
-        state = {"_hist": self._hist, "metadata": self.metadata}
+        """
+        Version 0.8: metadata added
+        Version 0.11: version added and set to 0. metadata/_hist replaced with dict.
+
+        ``dict`` contains __dict__ without "axes" and "_hist"
+        """
+        local_dict = copy.copy(self.__dict__)
+        del local_dict["axes"]
+        state = {"dict": local_dict, "version": 0}
         return state
```

Comment:
state should be a tuple rather than a dict.

Reply:
Okay, can do that, is a bit simpler. I think dicts are common in Python for state, but that works too.

Reply:
It also allows for a clean cut between old and new state format.

Reply:
I can't provide references either, other than that the default state is the `__dict__`, which is, well, a dict ;). But tuples work well here for us, for the reasons you gave above.

Reply:
I think pybind11 heavily favors tuples.